src/openms/source/ANALYSIS/MAPMATCHING/TransformationModel.cpp

```diff
@@ -60,5 +64,201 @@ namespace OpenMS
   {
     params.clear();
   }
+  
+  void TransformationModel::weightData(TransformationModel::DataPoints& data, const Param& params)
```

Comment:
I think the general interface would not pass the parameters into the function but rather the idea is to do this through `setParameters` and you will have to update the member variables in `updateMembers_`